kernel/pipes.c

```diff
@@ -173,16 +173,19 @@ static inline int z_vrfy_k_pipe_alloc_init(struct k_pipe *pipe, size_t size)
 #include <syscalls/k_pipe_alloc_init_mrsh.c>
 #endif
 
-void k_pipe_cleanup(struct k_pipe *pipe)
+int k_pipe_cleanup(struct k_pipe *pipe)
 {
-	__ASSERT_NO_MSG(!z_waitq_head(&pipe->wait_q.readers));
-	__ASSERT_NO_MSG(!z_waitq_head(&pipe->wait_q.writers));
+	if (z_waitq_head(&pipe->wait_q.readers) ||
+			z_waitq_head(&pipe->wait_q.writers)) {
```

Comment:
can you make it boolean operations ? e.g `z_waitq_head(&pipe->wait_q.readers) != NULL`